test/e2e/tests/security-provider.spec.js

```diff
@@ -0,0 +1,255 @@
+  let testApp;
+
+  async function mockSecurityProviderDetection(mockServer, scenario) {
+    switch (scenario) {
+      case 'notMalicious':
+        await mockServer.forPost(OPENSEA_URL).thenCallback(() => {
+          return {
+            statusCode: 200,
+            json: {
+              flagAsDangerous: 0,
+            },
+          };
+        });
+        break;
+      case 'malicious':
+        await mockServer.forPost(OPENSEA_URL).thenCallback(() => {
+          return {
+            statusCode: 200,
+            json: {
+              flagAsDangerous: 1,
+              reason:
+                'If you sign this request, you may lose all of your assets for good',
+              reason_header: 'This could be a scam',
+            },
+          };
+        });
+        break;
+      case 'notSafe':
+        await mockServer.forPost(OPENSEA_URL).thenCallback(() => {
+          return {
+            statusCode: 200,
+            json: {
+              flagAsDangerous: 2,
+              reason:
+                'The security provider didn’t detect any known malicious activity, but it still may not be safe to continue.',
+              reason_header: 'Request may not be safe',
+            },
+          };
+        });
+        break;
+      case 'requestNotVerified':
+        await mockServer.forPost(OPENSEA_URL).thenCallback(() => {
+          return {
+            statusCode: 401,
+            json: {
+              flagAsDangerous: 3,
+              reason:
+                'Because of an error, this request was not verified by the security. Proceed with caution.',
+              reason_header: 'Request not verified',
+            },
+          };
+        });
+        break;
+      default:
+        throw new Error(`Unknown scenario: ${scenario}`);
+    }
+  }
+
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+
+  it('Should return malicious response', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            transactionSecurityCheckEnabled: true,
+          })
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+        testSpecificMock: async (mockServer) =>
+          await mockSecurityProviderDetection(mockServer, 'malicious'),
+        dapp: true,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```

Comment:
same here, I think we can remove this line as it's not needed anymore

Reply:
Done